src/transforms/babel.js

```diff
@@ -26,19 +21,81 @@ module.exports = async function(asset) {
   }
 };
 
-async function shouldTransform(asset) {
+async function getConfig(asset) {
+  // If asset is marked as an ES6 modules, this is a second pass after dependencies are extracted.
+  // Just compile modules to CommonJS.
   if (asset.isES6Module) {
-    return true;
+    return {
+      plugins: [require('babel-plugin-transform-es2015-modules-commonjs')]
+    };
+  }
+
+  if (asset.babelConfig) {
+    return asset.babelConfig;
+  }
+
+  // Attempt to generate a config
+  let config = await getEnvConfig(asset);
+
+  // If this is the app module, and there is a .babelrc, merge it into the config.
+  // Don't load .babelrc for node_modules. https://github.com/parcel-bundler/parcel/issues/13.
+  if (!asset.name.includes('/node_modules/')) {
+    let babelrc =
+      (asset.package && asset.package.babel) ||
+      (await asset.getConfig(['.babelrc', '.babelrc.js']));
+    if (babelrc) {
+      config = Object.assign({}, babelrc, {
+        plugins: (babelrc.plugins || []).concat(config ? config.plugins : [])
+      });
+    }
+  }
+
+  return config;
+}
+
+module.exports.getConfig = getConfig;
+
+async function getEnvConfig(asset) {
+  // Load supported source engines (node and browser versions) for source module and target app,
+  // and generate a babel-preset-env config for each.
+  let sourceEngines = await getTargetEngines(asset);
+  let targetEngines = await getTargetEngines(asset, asset.options.mainFile);
```

Comment:
We should cache this somehow so we don't recompute the target engines for every single file